From e66a6c0301631cf37788eb3050e7b63682b46324 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sun, 12 Apr 2020 02:40:56 +0300
Subject: [PATCH 124/133] add QS Alpha overlays

	modified:   packages/SystemUI/res/color/qs_background_dark.xml
	modified:   packages/SystemUI/res/values/dimens.xml
	modified:   packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
	modified:   packages/overlays/Android.mk
	new file:   packages/overlays/QSPanelAlpha55Overlay/Android.mk
	new file:   packages/overlays/QSPanelAlpha55Overlay/AndroidManifest.xml
	new file:   packages/overlays/QSPanelAlpha55Overlay/res/values/dimens.xml
	new file:   packages/overlays/QSPanelAlpha65Overlay/Android.mk
	new file:   packages/overlays/QSPanelAlpha65Overlay/AndroidManifest.xml
	new file:   packages/overlays/QSPanelAlpha65Overlay/res/values/dimens.xml
	new file:   packages/overlays/QSPanelAlpha75Overlay/Android.mk
	new file:   packages/overlays/QSPanelAlpha75Overlay/AndroidManifest.xml
	new file:   packages/overlays/QSPanelAlpha75Overlay/res/values/dimens.xml
	new file:   packages/overlays/QSPanelAlpha85Overlay/Android.mk
	new file:   packages/overlays/QSPanelAlpha85Overlay/AndroidManifest.xml
	new file:   packages/overlays/QSPanelAlpha85Overlay/res/values/dimens.xml
---
 .../SystemUI/res/color/qs_background_dark.xml |  2 +-
 packages/SystemUI/res/values/dimens.xml       |  3 ++
 .../systemui/theme/ThemeOverlayManager.java   |  9 +++++-
 packages/overlays/Android.mk                  |  4 +++
 .../overlays/QSPanelAlpha55Overlay/Android.mk | 30 +++++++++++++++++++
 .../QSPanelAlpha55Overlay/AndroidManifest.xml | 24 +++++++++++++++
 .../res/values/dimens.xml                     | 21 +++++++++++++
 .../overlays/QSPanelAlpha65Overlay/Android.mk | 30 +++++++++++++++++++
 .../QSPanelAlpha65Overlay/AndroidManifest.xml | 24 +++++++++++++++
 .../res/values/dimens.xml                     | 21 +++++++++++++
 .../overlays/QSPanelAlpha75Overlay/Android.mk | 30 +++++++++++++++++++
 .../QSPanelAlpha75Overlay/AndroidManifest.xml | 24 +++++++++++++++
 .../res/values/dimens.xml                     | 21 +++++++++++++
 .../overlays/QSPanelAlpha85Overlay/Android.mk | 30 +++++++++++++++++++
 .../QSPanelAlpha85Overlay/AndroidManifest.xml | 24 +++++++++++++++
 .../res/values/dimens.xml                     | 21 +++++++++++++
 16 files changed, 316 insertions(+), 2 deletions(-)
 create mode 100644 packages/overlays/QSPanelAlpha55Overlay/Android.mk
 create mode 100644 packages/overlays/QSPanelAlpha55Overlay/AndroidManifest.xml
 create mode 100644 packages/overlays/QSPanelAlpha55Overlay/res/values/dimens.xml
 create mode 100644 packages/overlays/QSPanelAlpha65Overlay/Android.mk
 create mode 100644 packages/overlays/QSPanelAlpha65Overlay/AndroidManifest.xml
 create mode 100644 packages/overlays/QSPanelAlpha65Overlay/res/values/dimens.xml
 create mode 100644 packages/overlays/QSPanelAlpha75Overlay/Android.mk
 create mode 100644 packages/overlays/QSPanelAlpha75Overlay/AndroidManifest.xml
 create mode 100644 packages/overlays/QSPanelAlpha75Overlay/res/values/dimens.xml
 create mode 100644 packages/overlays/QSPanelAlpha85Overlay/Android.mk
 create mode 100644 packages/overlays/QSPanelAlpha85Overlay/AndroidManifest.xml
 create mode 100644 packages/overlays/QSPanelAlpha85Overlay/res/values/dimens.xml

diff --git a/packages/SystemUI/res/color/qs_background_dark.xml b/packages/SystemUI/res/color/qs_background_dark.xml
index c19fa080324..e06a4db871e 100644
--- a/packages/SystemUI/res/color/qs_background_dark.xml
+++ b/packages/SystemUI/res/color/qs_background_dark.xml
@@ -15,6 +15,6 @@
 -->
 
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
-    <item android:alpha="0.87"
+    <item android:alpha="@dimen/qs_panel_background_alpha"
           android:color="?android:attr/colorBackgroundFloating"/>
 </selector>
diff --git a/packages/SystemUI/res/values/dimens.xml b/packages/SystemUI/res/values/dimens.xml
index f151de81485..9f337303508 100644
--- a/packages/SystemUI/res/values/dimens.xml
+++ b/packages/SystemUI/res/values/dimens.xml
@@ -1165,4 +1165,7 @@
 
     <!-- Size of the RAT type for CellularTile -->
     <dimen name="celltile_rat_type_size">10sp</dimen>
+
+    <!-- The alpha of QS panel background -->
+    <item name="qs_panel_background_alpha" format="float" type="dimen">1</item>
 </resources>
diff --git a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
index 204bf2da88b..94d1e492ecf 100644
--- a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
+++ b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
@@ -76,6 +76,9 @@ class ThemeOverlayManager {
     @VisibleForTesting
     static final String OVERLAY_CATEGORY_DARKUI =
             "android.theme.customization.darkui";
+    @VisibleForTesting
+    static final String OVERLAY_CATEGORY_QSALPHA =
+            "android.theme.customization.qsalpha";
 
     /*
      * All theme customization categories used by the system, in order that they should be applied,
@@ -89,6 +92,7 @@ class ThemeOverlayManager {
             OVERLAY_CATEGORY_STATUSBAR,
             OVERLAY_CATEGORY_QSPANEL,
             OVERLAY_CATEGORY_DARKUI,
+            OVERLAY_CATEGORY_QSALPHA,
             OVERLAY_CATEGORY_ICON_ANDROID,
             OVERLAY_CATEGORY_ICON_SYSUI,
             OVERLAY_CATEGORY_ICON_SETTINGS,
@@ -102,6 +106,7 @@ class ThemeOverlayManager {
             OVERLAY_CATEGORY_SHAPE,
             OVERLAY_CATEGORY_QSPANEL,
             OVERLAY_CATEGORY_STATUSBAR,
+            OVERLAY_CATEGORY_QSALPHA,
             OVERLAY_CATEGORY_DARKUI,
             OVERLAY_CATEGORY_ICON_ANDROID,
             OVERLAY_CATEGORY_ICON_SYSUI);
@@ -125,7 +130,8 @@ class ThemeOverlayManager {
                 OVERLAY_CATEGORY_COLOR, OVERLAY_CATEGORY_FONT, OVERLAY_CATEGORY_STATUSBAR,
                 OVERLAY_CATEGORY_SHAPE, OVERLAY_CATEGORY_ICON_ANDROID, OVERLAY_CATEGORY_DARKUI));
         mTargetPackageToCategories.put(SYSUI_PACKAGE,
-                Sets.newHashSet(OVERLAY_CATEGORY_ICON_SYSUI, OVERLAY_CATEGORY_QSPANEL));
+                Sets.newHashSet(OVERLAY_CATEGORY_ICON_SYSUI, OVERLAY_CATEGORY_QSPANEL,
+								OVERLAY_CATEGORY_QSALPHA));
         mTargetPackageToCategories.put(SETTINGS_PACKAGE,
                 Sets.newHashSet(OVERLAY_CATEGORY_ICON_SETTINGS));
         mTargetPackageToCategories.put(mLauncherPackage,
@@ -139,6 +145,7 @@ class ThemeOverlayManager {
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_ANDROID, ANDROID_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_DARKUI, ANDROID_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_QSPANEL, SYSUI_PACKAGE);
+        mCategoryToTargetPackage.put(OVERLAY_CATEGORY_QSALPHA, SYSUI_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_SYSUI, SYSUI_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_SETTINGS, SETTINGS_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_LAUNCHER, mLauncherPackage);
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index 50617297575..7793be94d67 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -83,6 +83,10 @@ LOCAL_REQUIRED_MODULES := \
 	QSPanel55Overlay \
 	CharcoalDarkUIOverlay \
 	MidnightDarkUIOverlay \
+	QSPanelAlpha55Overlay \
+	QSPanelAlpha65Overlay \
+	QSPanelAlpha75Overlay \
+	QSPanelAlpha85Overlay \
 	NavigationBarMode3ButtonOverlay \
 	NavigationBarMode2ButtonOverlay \
 	NavigationBarModeGesturalOverlay \
diff --git a/packages/overlays/QSPanelAlpha55Overlay/Android.mk b/packages/overlays/QSPanelAlpha55Overlay/Android.mk
new file mode 100644
index 00000000000..6d1455dec05
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha55Overlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := QSPanelAlpha55
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := QSPanelAlpha55Overlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/QSPanelAlpha55Overlay/AndroidManifest.xml b/packages/overlays/QSPanelAlpha55Overlay/AndroidManifest.xml
new file mode 100644
index 00000000000..44e79dbe1f4
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha55Overlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.qsalpha55"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="com.android.systemui" android:category="android.theme.customization.qsalpha" android:priority="1"/>
+    <application android:label=".55" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/QSPanelAlpha55Overlay/res/values/dimens.xml b/packages/overlays/QSPanelAlpha55Overlay/res/values/dimens.xml
new file mode 100644
index 00000000000..288fa27a9b9
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha55Overlay/res/values/dimens.xml
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <item name="qs_panel_background_alpha" format="float" type="dimen">0.55</item>
+</resources>
diff --git a/packages/overlays/QSPanelAlpha65Overlay/Android.mk b/packages/overlays/QSPanelAlpha65Overlay/Android.mk
new file mode 100644
index 00000000000..fc2a1d8c67e
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha65Overlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := QSPanelAlpha65
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := QSPanelAlpha65Overlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/QSPanelAlpha65Overlay/AndroidManifest.xml b/packages/overlays/QSPanelAlpha65Overlay/AndroidManifest.xml
new file mode 100644
index 00000000000..b381f609834
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha65Overlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.qsalpha65"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="com.android.systemui" android:category="android.theme.customization.qsalpha" android:priority="1"/>
+    <application android:label=".65" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/QSPanelAlpha65Overlay/res/values/dimens.xml b/packages/overlays/QSPanelAlpha65Overlay/res/values/dimens.xml
new file mode 100644
index 00000000000..8d466b49e30
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha65Overlay/res/values/dimens.xml
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <item name="qs_panel_background_alpha" format="float" type="dimen">0.65</item>
+</resources>
diff --git a/packages/overlays/QSPanelAlpha75Overlay/Android.mk b/packages/overlays/QSPanelAlpha75Overlay/Android.mk
new file mode 100644
index 00000000000..b8ddf6f2549
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha75Overlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := QSPanelAlpha75
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := QSPanelAlpha75Overlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/QSPanelAlpha75Overlay/AndroidManifest.xml b/packages/overlays/QSPanelAlpha75Overlay/AndroidManifest.xml
new file mode 100644
index 00000000000..7d8f56f0a19
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha75Overlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.qsalpha75"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="com.android.systemui" android:category="android.theme.customization.qsalpha" android:priority="1"/>
+    <application android:label=".75" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/QSPanelAlpha75Overlay/res/values/dimens.xml b/packages/overlays/QSPanelAlpha75Overlay/res/values/dimens.xml
new file mode 100644
index 00000000000..fc3e3faf264
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha75Overlay/res/values/dimens.xml
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <item name="qs_panel_background_alpha" format="float" type="dimen">0.75</item>
+</resources>
diff --git a/packages/overlays/QSPanelAlpha85Overlay/Android.mk b/packages/overlays/QSPanelAlpha85Overlay/Android.mk
new file mode 100644
index 00000000000..d263adba673
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha85Overlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := QSPanelAlpha85
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := QSPanelAlpha85Overlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/QSPanelAlpha85Overlay/AndroidManifest.xml b/packages/overlays/QSPanelAlpha85Overlay/AndroidManifest.xml
new file mode 100644
index 00000000000..2787be2e4ac
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha85Overlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.qsalpha85"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="com.android.systemui" android:category="android.theme.customization.qsalpha" android:priority="1"/>
+    <application android:label=".85" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/QSPanelAlpha85Overlay/res/values/dimens.xml b/packages/overlays/QSPanelAlpha85Overlay/res/values/dimens.xml
new file mode 100644
index 00000000000..9fe20bf71bb
--- /dev/null
+++ b/packages/overlays/QSPanelAlpha85Overlay/res/values/dimens.xml
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <item name="qs_panel_background_alpha" format="float" type="dimen">0.85</item>
+</resources>
-- 
2.17.1

